File: Lab8/ModifiedMVCExample/ModifiedMVCModel/src/Controller/StudentController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Controller;

import Model.Student;
import View.StudentView;

/**
 *
 * @author Ghani
 */
public class StudentController {
    Student student ;// = new Student("Rehman Ghani","FA22-BSE-099",5,3.84);
    StudentView stdView;// = new StudentView();
    public StudentController(Student student, StudentView stdView){
        this.stdView = stdView;
        this.student = student;
    }
    public void setData(Student std){
       Student stud = new Student(student.getName(),student.getRegNo(),student.getSemister(),student.getGpa());
    }
    public Student getData(){
       return student;
    }
    public void updateStudentView(){
        stdView.showStudentDetails(student);
    }
    
}

File: Lab8/MVCInPython/InPython/controller/student_controller.py
# controller/student_controller.py

from model.student import Student
from view.student_view import StudentView

class StudentController:
    def __init__(self, student, view):
        self._student = student
        self._view = view

    # Update student details
    def set_data(self, name=None, reg_no=None, semester=None, gpa=None):
        if name is not None:
            self._student.set_name(name)
        if reg_no is not None:
            self._student.set_reg_no(reg_no)
        if semester is not None:
            self._student.set_semester(semester)
        if gpa is not None:
            self._student.set_gpa(gpa)

    # Get student details
    def get_data(self):
        return self._student

    # Update view
    def update_student_view(self):
        self._view.show_student_details(self._student)